### PR TITLE
Add type definitions for mui-datatables

### PR DESCRIPTION
**Pull request by favna**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project. ==> Based on [this gist](https://gist.github.com/nidheeshdas/ebdb07d147d7ff89e09bb44fe822ba31) as well as dissecting the package myself / reading what is documented all the while following the DefinitelyTyped common type declaration methodologies
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mui-datatables/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+export interface MUIDataTableProps {
+    title: string;
+    columns: MUIDataTableColumnDef[];
+    data: any[];
+    options?: MUIDataTableOptions;
+}
+
+export const MUIDataTable: React.ComponentType<MUIDataTableProps>;
```
Comment (PranavSenthilnathan) [inline types/mui-datatables/index.d.ts]: This export makes it seem like I should be able to do:
`import { MUIDataTable } from 'mui-datatables';`

But I don't see this export in the library. If it is, then adding it to the test would be great.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

**Review by PranavSenthilnathan — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Comment by gabejohnson** [comment]
@Favna are you planning to add types for the submodules exported in https://github.com/gregnb/mui-datatables/pull/314? If so, let me know if you need any help.

**Comment by Favna** [comment]
@gabejohnson I see another update is in order due to the merging of https://github.com/gregnb/mui-datatables/pull/434 so might as well cover the subcomponents while I do that. I haven't delved into it just yet but if you could provide some ready-made information on the new changes that would most definitely help. Since GitHub isn't exactly a medium for conversation like that - could you hit me up on either Twitter (`@Favna_`) or Discord (`Favna#001`)?

**Pull request merged**